src/commands.ts

```diff
@@ -1706,6 +1710,45 @@ export const insertTableFootnote = (
   dispatch(view.state.tr.setSelection(textSelection).scrollIntoView())
 }
 
+export const insertBoxElement = (
+  state: ManuscriptEditorState,
+  dispatch?: Dispatch
+) => {
+  const selection = state.selection
+
+  // Check if the selection is inside the body
+  const isBody = hasParentNodeOfType(schema.nodes.body)(selection)
+  const isBoxText = hasParentNodeOfType(schema.nodes.box_element)(selection)
```

Comment:
@asouqi 
can you please review this ? 
I made this change to prevent adding a box text inside another.